### DISCUSSION TITLE
Mutation isLoading changed to false after callbacks

### DISCUSSION THREAD

**Discussion by egorovsa** [TanStack/query#3721] [General] [answered]
I've faced a problem that after mutation isLoading got back to false after the onSuccess callback fired. However to my mind it should be changed before.
For instance onSuccess redirects us to another page by router and component is unmounted before  than isLoading changed to false. 

example
```
 const {
    isLoading, 
    data,
    mutation
  } = useMyFooQuery();

   function goBackToList(){
      // isLoading still true, but should be false
     // redirect ////
  }

   mutation.mutate(updatedData, {
      onSuccess: goBackToList,
   });

//....

```

Any idea ?

**Comment by TkDodo** [comment, marked as answer]
`isLoading` that was desctructured from `useMyFooQuery` will update on the next re-render, so no, you can't read it in callbacks (that are executed _before_ the re-render) and expect to have changed.

**Reply by egorovsa** [reply]
Yes. Thanks. It's clear.
